core/src/main/scala/com/salesforce/op/ModelInsights.scala

```diff
@@ -647,6 +693,32 @@ case object ModelInsights {
     }
   }
 
+  private[op] def descaleLRContrib(
+    model: Option[Model[_]],
+    sparkFtrContrib: Seq[Double],
+    featureStd: Double,
+    labelStd: Double): Option[Seq[Double]] = {
+    val stage = model.flatMap {
+      case m: SparkWrapperParams[_] => m.getSparkMlStage()
+      case _ => None
+    }
+    stage.collect {
+      case m: LogisticRegressionModel =>
+        if (m.getStandardization && sparkFtrContrib.nonEmpty) {
+          // scale entire feature contribution vector
+          sparkFtrContrib.map(_ * featureStd)
```

Comment:
can you explain the logic to me here? the spark coefficients are on the actual scale of the data. If a feature has a large standard deviation that means it has a large range, if it has a small standard deviation it has a small range. Shouldn't you be dividing by the standard deviation to bring them all into a range that can be compared? 
Is there something I am missing?

Reply:
It's a bit counter-intuitive, and only works for certain cases. For regression let's take `y = 1 mil * x_1 + 1 * x_2`, where `x_1` is small scale and `x_2` is large scale (millions), and due to the formula `y` will be on the same large scale as `x_2`. 

Spark in this case will return (1 mil, 1) but instead we want roughly both coeff to be between 0 and 1. The small feature have a small range, so 1mil * std(small feature) / std(y -- large) will bring it back to a small scale, because it is multiplied by a small number and divided by a large number. The large feature 1 * std (large feature) / std(y -- large) will stay roughly the same, because it is multiplied by a large number and divided by a large number. 

The same math is harder to reason for the classification case, because y is either 0, 1 and we are regressing on odds. p (probability y = 1) = 1 / 1 + e^ -( 1 mil * x_1 + 1 * x_2) then spark is still returning (1mil, 1) and by descaling, we instead returns (1, 1 mil) ?

Reply:
Can you add a citation as a comment for future reference?